### PR TITLE
Feat/fluxes

### LINKED ISSUE #17: add goodtables schema for proteomics
done when we can upload data to proteomics and fluxomics service

### PR DESCRIPTION
**Pull request by hredestig**
- support uploading omics data
- fix #17

### CODE REVIEW

File: upload/checks.py [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+from upload import iloop_client, logger
+from upload.settings import Default
+
+
+IDENTIFIER_TYPES = {'protein', 'reaction'}
+
+
+def load_identifiers(type):
+    iloop = iloop_client(Default.ILOOP_API, Default.ILOOP_TOKEN)
+    return set(iloop.BiologicalIdentifier.subset(type=type))
```
Comment (Midnighter) [inline upload/checks.py]: Could make this and the `IDENTIFIER_TYPES` plus `UPLOAD_TYPES` `frozenset`s so they can't be modified but it's certainly not necessary.

File: upload/settings.py [hunk 2]
```diff
@@ -1,17 +1,8 @@
...
-ILOOP_API = os.environ.get('ILOOP_API') or config.get('defaults', 'api', fallback=None)
-ILOOP_TOKEN = os.environ.get('ILOOP_TOKEN') or config.get('defaults', 'token', fallback=None)
 
 
 class Default(object):
-    ILOOP_API = ILOOP_API
-    ILOOP_TOKEN = ILOOP_TOKEN
+    ILOOP_API = os.environ.get('ILOOP_API')
+    ILOOP_TOKEN = os.environ.get('ILOOP_TOKEN')
     ILOOP_BIOSUSTAIN = 'https://iloop.biosustain.dtu.dk/api'
```
Comment (Midnighter) [inline upload/settings.py]: I don't understand immediately why we potentially would like to connect to two different running iLoop instances.

File: upload/checks.py (hunk 1 above)
Comment (hredestig) [inline upload/checks.py]: sure, done

**Comment by hredestig** [comment]
good point on using one instead of first, fixed

File: upload/settings.py (hunk 2 above)
Comment (hredestig) [inline upload/settings.py]: we have cfb iloop and the decaf one..

Comment (hredestig) [inline upload/settings.py]: but agree, this implementation here is not so pretty. Let's handle in different pr though as not related to the flux topic

**Comment by hredestig** [comment]
Thanks for the review, I need to release these fixes so will go ahead and merge, please raise issue if there was still something left to do

**Comment by Midnighter** [comment]
Don't think it's worth creating issues but two points we should keep in the back of our heads:

1. The `iloop.Model.one` method may raise `NoResultFound` or `MultipleResultsFound` (not sure if/how those exceptions are wrapped by Potion). We may need to catch those exceptions if they bubble up to the frontend somehow.
2. Using the `namedtuple` for the `PROJECT_OBJECT` is a nice way of going about but it's also the perfect use-case for a mocked instance. Might consider using [factoryboy](https://github.com/pytest-dev/pytest-factoryboy) in future.

**Pull request merged**